app/src/main/java/app/revanced/manager/MainActivity.kt

```diff
@@ -51,9 +60,45 @@ class MainActivity : ComponentActivity() {
 
                 NavBackHandler(navController)
 
-                val showAutoUpdatesDialog by vm.prefs.showAutoUpdatesDialog.getAsState()
-                if (showAutoUpdatesDialog) {
-                    AutoUpdatesDialog(vm::applyAutoUpdatePrefs)
+                val firstLaunch by vm.prefs.firstLaunch.getAsState()
+
+                if (firstLaunch) {
+                    val showAutoUpdatesDialog = rememberSaveable { mutableStateOf(false) }
+                    if (!showAutoUpdatesDialog.value) {
+                        val launcher = rememberLauncherForActivityResult(
+                            contract = ActivityResultContracts.StartActivityForResult()
+                        ) { result: ActivityResult ->
+                            if (result.resultCode == RESULT_OK) {
+                                if (result.data != null) {
+                                    val jsonData = result.data!!.getStringExtra("data")!!
+                                    vm.applyLegacySettings(jsonData)
+                                }
+                            } else {
+                                showAutoUpdatesDialog.value = true
+                                toast(getString(R.string.legacy_import_failed))
+                            }
+                        }
+
+                        val intent = Intent().apply {
+                            setClassName(
+                                "app.revanced.manager.flutter",
+                                "app.revanced.manager.flutter.ExportSettingsActivity"
+                            )
+                        }
+
+                        LaunchedEffect(Unit) {
+                            try {
+                                launcher.launch(intent)
```

Comment:
The `LaunchedEffect` block will run again if you rotate the screen for example. Add a variable to keep track if you have attempted to launch or not if this is not intended